webapp/src/oEmbed.jsx

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import {id as pluginId} from './manifest';
+import {getDimensionsForProvider, getOEmbedUrl, getProviderForUrl} from './utils';
+
+export default class OEmbed extends Component {
+    static propTypes = {
+        embed: PropTypes.shape({
+            url: PropTypes.string.isRequired,
+        }),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.oembedUrl = getOEmbedUrl(this.props.embed.url);
+        const provider = getProviderForUrl(this.props.embed.url);
+        this.dimensions = getDimensionsForProvider(provider.provider_name);
+        this.state = {
+            html: null,
+            thumbnailUrl: null,
+        };
+    }
+
+    async componentWillMount() {
+        try {
+            const res = await (await fetch(`/plugins/${pluginId}/`, {
+                method: 'POST',
+                headers: {'Content-Type': 'application/json'},
+                body: this.oembedUrl,
+            })).json();
+
+            if (res.html) {
+                this.setState({html: res.html});
+            } else {
+                this.setState({thumbnailUrl: res.thumbnail_url});
+            }
+        } catch (err) {
+            // eslint-disable-next-line no-console
+            console.error(err);
+        }
+    }
+
+    render() {
+        const {html, thumbnailUrl} = this.state;
+        const style = {
+            width: `${this.dimensions.width}px`,
```

Comment:
`getDimensionsForProvider` can return null and you are not handling that here

Reply:
`getDimensionsForProvider` should never return `null` at this point because [a precondition for the matcher function is that the provider should have dimensions registered](https://github.com/mattermost/mattermost-oembed-plugin/pull/1/files#diff-16d51f9caeee685540847c205f4e784dR16).

Falling back at this point is not possible (hence the check on the matcher) but I can add a check for `null` on the render function to be 100% safe.